python/cudf/cudf/core/index.py

```diff
@@ -466,8 +487,36 @@ def __init__(self, start, stop=None, name=None):
             start, stop = 0, start
         self._start = int(start)
         self._stop = int(stop)
-        self.name = name
         self._cached_values = None
+        self._index = None
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    def as_column(self):
+        if len(self) > 0:
+            vals = cudautils.arange(self._start, self._stop, dtype=self.dtype)
+        else:
+            vals = rmm.device_array(0, dtype=self.dtype)
+        return column.build_column(data=Buffer(vals), dtype=vals.dtype,)
+
+    @property
+    def _values(self):
+        if self._cached_values is None:
+            self._cached_values = self.as_column()
+        return self._cached_values
+
+    @property
+    def _data(self):
+        from cudf.utils.utils import OrderedColumnDict
+
+        return OrderedColumnDict({self.name: self._values})
```

Comment:
Does this need to be special cased for `RangeIndex`?

Reply:
Unlike other subclasses of `Table`, `RangeIndex`'s constructor doesn't call `super().__init__` to initialize `_data`.

Instead we wait till someone needs `._data` and only then compute (and cache) it.